portfolio/src/main/webapp/calendar.js

```diff
@@ -69,14 +103,70 @@ function getTimeObject(timeString) {
   return new Date(currentYear, currentMonth, currentDate, timeHour, timeMinute);
 }
 
-/** Creates a single element for a new calendar event. */
+/** Creates a card element for a new calendar event. */
+/** 
+ * TODO(hollyyuqizheng) Look into refactoring the card creation code 
+ * into a JavaScript class. This card element will be used in different 
+ * sections on the UI, so this refactoring will make things simpler. 
+ */
 function updateCalendarEventList(newCalendarEvent) {
-  const newEventElement = document.createElement('li');
-  newEventElement.innerHTML = 'New Event: ' + newCalendarEvent.name +
-      '<br>starts at ' + newCalendarEvent.startTime + '<br>ends at ' +
-      newCalendarEvent.endTime;
+  const newEventCard = document.createElement('div');
+  newEventCard.classList.add('card');
+
+  const cardBody = document.createElement('div');
+  cardBody.classList.add('card-body');
+  newEventCard.appendChild(cardBody);
+
+  const cardTitle = document.createElement('h4');
+  cardTitle.classList.add('card-title');
+  cardTitle.innerText = newCalendarEvent.name;
+  cardBody.appendChild(cardTitle);
+
+  const startTimeText = document.createElement('p');
+  startTimeText.classList.add('card-text');
+  startTimeText.innerText = newCalendarEvent.startTime;
+  cardBody.appendChild(startTimeText);
+
+  const endTimeText = document.createElement('p');
+  endTimeText.classList.add('card-text');
+  endTimeText.innerText = newCalendarEvent.endTime;
+  cardBody.appendChild(endTimeText);
+
+  const deleteButton = document.createElement('button');
+  deleteButton.classList.add('btn');
+  deleteButton.classList.add('btn-danger');
+  deleteButton.innerText = 'Delete this event';
+  cardBody.appendChild(deleteButton);
 
   const eventList = document.getElementById('new-event-list');
   eventList.innterHTML = '';
-  eventList.appendChild(newEventElement);
+  eventList.appendChild(newEventCard);
+
+  // The delete button removes the event's card from the UI.
+  deleteButton.onclick = function(newEventCard) {
+    newEventCard.target.closest('div.card').remove();
+  }
+}
+
+/** Collects and returns all the events currently displayed on the UI. */
```

Comment:
Can we indicate in the comment what it returns? Does it return an array of strings? JSONs?

I think it returns JSON strings, but I think it should return a list of CalendarEvent's instead. I think it should be up to the caller how they want to handle it further.

Reply:
Updated. Now this function returns an array of CalendarEvent's.